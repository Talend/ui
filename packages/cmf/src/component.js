import invariant from 'invariant';
import actionCreator from './actionCreator';
import expression from './expression';
import registry from './registry';
import CONST from './constant';

/**
 * All stuff related to the routing in CMF
 * @module react-cmf/lib/component
 * @see module:react-cmf/lib/component
 */

/**
 * return a component from the registry
 * @param  {string} id the component id you want
 * @param  {object} context optional context to get the registry
 * @return {function} the react component
 */
function get(id, context) {
	const component = registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context);
	if (!component) {
		throw new Error(`component not found in the registry: ${id}`);
	}
	return component;
}

/**
 * register a component for the router configuration
 * @param  {string} id the component id you want to register
 * @param  {any} component the component you want to register
 * @param  {object} context optional context to get the registry
 */
function register(id, component, context) {
	if (!component) {
		invariant(process.env.NODE_ENV !== 'production', 'You can register undefined as a component');
		return;
	}
	registry.addToRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, component, context);
	if (component.actions) {
		Object.keys(component.actions).forEach(key => {
			actionCreator.register(key, component.actions[key], context);
		});
	}
	if (component.expressions) {
		Object.keys(component.expressions).forEach(key => {
			expression.register(key, component.expressions[key], context);
		});
	}
}

const registerMany = registry.getRegisterMany(register);

function has(id, context) {
	return (
		registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context) !== undefined
	);
}

export default {
	get,
	has,
	register,
	registerMany,
};
