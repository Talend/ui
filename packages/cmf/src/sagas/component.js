import invariant from 'invariant';
import { fork, cancel, take, takeEvery } from 'redux-saga/effects';
import curry from 'lodash/curry';
import CONST from '../constant';
import registry from '../registry';

/**
 * This function register a saga in the cmf registry
 * @param {string} id the saga id you want
 * @param {generator} saga the saga generator
 * @param {object} context optional context to get the registry
 */
export function register(id, saga, context) {
	registry.addToRegistry(`SAGA:${id}`, saga, context);
}

/**
 * This function allow to get a saga from the registry
 * @param {string} id the saga id you want
 * @param {object} context optional context to get the registry
 */
export function get(id, context) {
	return registry.getFromRegistry(`SAGA:${id}`, context);
}

export const registerMany = registry.getRegisterMany(register);

export const isActionCancelable = curry(
	(startAction, action) =>
		action.type === `${CONST.WILL_UNMOUNT_SAGA_STOP}_${startAction.saga}` &&
		startAction.event.componentId === action.event.componentId,
);

export function* onSagaStart(action) {
	const saga = get(action.saga);
	if (!saga) {
		invariant(
			process.env.NODE_ENV === 'production',
			`You cannot register undefined as saga for id "${action.saga}"`,
		);
	} else {
		const task = yield fork(saga, action.props);
		yield take(isActionCancelable(action));
		yield cancel(task);
	}
}

export function* handle() {
	yield takeEvery(CONST.DID_MOUNT_SAGA_START, onSagaStart);
	yield take('DO_NOT_QUIT');
}
