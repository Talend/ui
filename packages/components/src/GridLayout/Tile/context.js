import React, { useContext } from 'react';
import invariant from 'invariant';

// eslint-disable-next-line import/prefer-default-export
export const TileContext = React.createContext();

export function useTileContext() {
	const context = useContext(TileContext);
	invariant(
		context,
		'@talend/react-components > Tile: you are using a sub component out of Tile.Container.',
	);
	return context;
}
