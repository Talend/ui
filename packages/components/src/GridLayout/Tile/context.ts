import { createContext, useContext } from 'react';

export type TileContextType = {
	displayMode: string;
	setDisplayMode: (mode: string) => void;
};

export const TileContext = createContext<TileContextType>({
	displayMode: '',
	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	setDisplayMode: (mode: string) => {},
});

export function useTileContext() {
	const context = useContext(TileContext);
	return context;
}
