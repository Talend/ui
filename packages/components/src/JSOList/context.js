import React from 'react';

export const ListContext = React.createContext();
