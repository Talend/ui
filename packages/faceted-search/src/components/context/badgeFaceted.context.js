import { createContext, useContext } from 'react';
import invariant from 'invariant';

const badgeFacetedContext = createContext();

const useBadgeFacetedContext = () => {
	const context = useContext(badgeFacetedContext);
	invariant(
		context,
		'[badgeFacetedContext]: You are using a badge faceted component outside its context',
	);
	return context;
};

const BadgeFacetedProvider = badgeFacetedContext.Provider;

export { BadgeFacetedProvider, useBadgeFacetedContext };
