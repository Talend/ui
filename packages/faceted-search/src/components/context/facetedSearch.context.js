import { createContext, useContext } from 'react';
import invariant from 'invariant';

const facetedSearchContext = createContext();

const useFacetedSearchContext = () => {
	const context = useContext(facetedSearchContext);
	invariant(
		context,
		'[facetedSearchContext]: You are using one of faceted mode outside the faceted search',
	);
	return context;
};

const FacetedSearchProvider = facetedSearchContext.Provider;

export { FacetedSearchProvider, useFacetedSearchContext };
