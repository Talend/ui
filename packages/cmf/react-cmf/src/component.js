import invariant from 'invariant';
import actionCreator from './actionCreator';
import expression from './expression';
import sagas from './sagas';
import registry from './registry';
import CONST from './constant';

/**
 * All stuff related to the routing in CMF
 * @module react-cmf/lib/component
 * @see module:react-cmf/lib/component
 */

/**
 * return a component from the registry
 * @param  {string} id the component id you want
 * @param  {object} context optional context to get the registry
 * @return {function} the react component
 */
function get(id, context) {
	const component = registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context);
	if (!component) {
		throw new Error(`component not found in the registry: ${id}`);
	}
	return component;
}

/**
 * register a component to let CMF be able to Inject it.
 * @param  {string} id the component id you want to register
 * @param  {any} component the component you want to register
 * @param  {object} context optional context to get the registry
 */
function register(id, component, context) {
	if (!component) {
		invariant(
			process.env.NODE_ENV === 'production',
			'You cannot register undefined as a component for id "%s"',
			id,
		);
		return;
	}
	registry.addToRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, component, context);
	if (component.actions) {
		actionCreator.registerMany(component.actions, context);
	}
	if (component.expressions) {
		expression.registerMany(component.expressions, context);
	}
	if (component.sagas) {
		sagas.registerMany(component.sagas, context);
	}
}

const registerMany = registry.getRegisterMany(register);

function has(id, context) {
	return (
		registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context) !== undefined
	);
}

export default {
	get,
	has,
	register,
	registerMany,
};
