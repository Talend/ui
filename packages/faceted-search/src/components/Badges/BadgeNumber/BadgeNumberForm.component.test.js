import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount } from 'enzyme';
import { BadgeNumberForm } from './BadgeNumberForm.component';
import { BadgeFacetedProvider } from '../../context/badgeFaceted.context';
import getDefaultT from '../../../translate';

const badgeFacetedContextValue = {
	onDeleteBadge: jest.fn(),
	onHideOperator: jest.fn(),
	onSubmitBadge: jest.fn(),
};

describe('BadgeNumberForm', () => {
	it('should mount a default badge', () => {
		// Given
		const props = {
			id: 'customId',
			onSubmit: jest.fn(),
			t: getDefaultT(),
		};
		// When
		const wrapper = mount(
			<BadgeFacetedProvider value={badgeFacetedContextValue}>
				<BadgeNumberForm {...props} />
			</BadgeFacetedProvider>,
		);
		// Then
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should mount a badge with some other values', () => {
		// Given
		const onSubmit = jest.fn();
		const props = {
			id: 'customId',
			onSubmit,
			value: 'i230982903',
			t: getDefaultT(),
		};
		// When
		const wrapper = mount(
			<BadgeFacetedProvider value={badgeFacetedContextValue}>
				<BadgeNumberForm {...props} />
			</BadgeFacetedProvider>,
		);
		// Then
		expect(
			wrapper
				.find('input[type="number"]')
				.first()
				.props().value,
		).toEqual('i230982903');

		const submitButton = wrapper.find('button[type="submit"]').first();
		submitButton.simulate('submit');

		expect(onSubmit).toHaveBeenCalled();
	});
});
