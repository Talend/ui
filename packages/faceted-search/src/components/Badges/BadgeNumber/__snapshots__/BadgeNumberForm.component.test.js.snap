// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeNumberForm should mount a default badge 1`] = `
<BadgeNumberForm
  id="customId"
  onSubmit={[MockFunction]}
  t={[Function]}
/>
`;
